src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1000,6 +1001,7 @@ public Module getModule() {
 
     private transient Object classData; // Set by VM
     private transient Object[] signers; // Read by VM, mutable
+    private final transient int modifiers;  // Set by the VM
```

Comment:
@coleenp Could this field be @Stable, or does that only apply to `putfield`s?

Reply:
I don't think this needs to be stable - finals in java.lang is trusted by the JIT compiler.

Reply:
Yeah, I was just thinking whether something set from inside the VM which is marked @Stable is constant-folded :)

Reply:
I don't think @Stable would hurt but final should provide the same guarantee.  It's set internally by the VM so there's no late setting. I don't know if this field implementation can constant fold in the case of Arrays which are (JVM_ACC_ABSTRACT | JVM_ACC_FINAL | JVM_ACC_PUBLIC).

Reply:
@viktorklang-ora `@Stable` is not about how the field was set, but about the JIT observing a  non-default value at compile time.  If it observes a non-default value, it can treat it as a compile time constant.

Reply:
@DanHeidinga Great explanation, thank you!